data-runtime/src/main/java/io/micronaut/data/runtime/convert/DataConversionServiceFactory.java

```diff
@@ -634,6 +634,9 @@ DataConversionServiceImpl build(@NonNull BeanContext beanContext) {
         // ZonedDateTime
         addZonedConvertorsConvertors(conversionService, ZonedDateTime.class, Function.identity());
 
+        // LocalDate
+        conversionService.addConverter(LocalDate.class, java.sql.Date.class, java.sql.Date::valueOf);
```

Comment:
Need to convert to `java.sql.Date` or else converter from core that converts to `UTC` will be used. And we need sql date anyway.